TACC/Frontera-Portal
# task/FP-740 - public url postit

## Overview: ##

Implementation of Links feature (public URL in CEPv1)

## PR Status: ##


## Related Jira tickets: ##

* [FP-740](https://jira.tacc.utexas.edu/browse/FP-740)

## Summary of Changes: ##

- Added end point and models for PublicUrl
- Added temporary client implementation for public URL
- Added TextCopyField component

## Testing Steps: ##
0. Rebuild core image: `make build`
1. Add "makeLink" to _WORKBENCH_SETTINGS: `"makeLink": True`
2. Select a single file
3. Click "Public URL"
4. Click the "Generate" button
5. Click the "Copy" button
6. See if you can redeem that link by pasting in your browser's url bar
7. Close the modal
8. Open the modal and see that it returns your previously generated URL

Delete and Replace Link are not currently functional due to an agavepy bug

## UI Photos:

![image](https://user-images.githubusercontent.com/17181582/102257855-e3277400-3ed2-11eb-9ad1-5ef4d7c02579.png)

![image](https://user-images.githubusercontent.com/17181582/102257888-ecb0dc00-3ed2-11eb-8f8b-670c2adf9375.png)

![image](https://user-images.githubusercontent.com/17181582/102257909-f4708080-3ed2-11eb-9374-24739b634f0a.png)

![image](https://user-images.githubusercontent.com/17181582/102257941-fc302500-3ed2-11eb-8751-36a7c5bac0e4.png)

## Notes: ##
